pyspecdata/core.py

```diff
@@ -5466,7 +5474,10 @@ def extend(self,axis,extent, fill_with=0, tolerance=1e-5):
             stop_index_addto = int((extent - u[-1]) // du)
             if ((extent - u[-1]) - du * stop_index_addto)/du > tolerance:# the first quantity here is negative
                 stop_index_addto += 1
+            if stop_index_addto > 100:
+                stop_index_addto = 1
             stop_index += stop_index_addto
+            logger.debug(stop_index)
         else:
             raise RuntimeError("extent ({:g}) needs to be further than the bounds on '{:s}', which are {:g} and {:g}".format(extent,axis,u[0],u[-1]))
```

Comment:
a general comment -- can you add a **very** short file to the examples directory (under the main root of the repo) in order to give an example of the two use cases that you refer to here?  I would suggest, e.g., constructing an nddata of random numbers and extending it.  Show an example like you tried to start with and then an example like the "other case" you refer to (I imagine these are two differently shaped nddata).

Reply:
so I've attempted this with `examples/extend.py` above.

Reply:
Yes, and this currently works as intended